server/src/yml-objects/AbstractYmlObject.ts

```diff
@@ -75,38 +84,69 @@ export abstract class AbstractYmlObject implements CompletionItem {
      * that are out of the current scope.
      */
     public enrichWith(
-        fields: FieldContext[],
-        connection: IConnection,
+        documentation: string,
+        type: string,
         sourceElementName?: string,
-        baseType?: string,
         scopeStartOffset?: number,
         scopeEndOffset?: number,
     ): void {
-        sourceElementName = sourceElementName ? sourceElementName : 'static';
+        this.sourceElementName = sourceElementName ?? 'static';
         this.data = `id_${sourceElementName}_${this.label}`;
-        this.setDocumentation(getDocumentation(fields, connection));
-        this.detail = getType(fields, connection, baseType);
+        this.setUserInformations(this.buildDetailString(type), documentation);
         if (scopeEndOffset && scopeStartOffset) {
             this.scopeEndOffset = scopeEndOffset;
             this.scopeStartOffset = scopeStartOffset;
         }
     }
 
+    protected buildDetailString(type: string): string {
+        const separator = this.sourceElementName === 'static' ? ' ' : '.';
+        return `(${this.kindName}) [${this.sourceElementName}]${separator}${this.label} ⇒ ${type}`;
+    }
+
+    public getHoverContent(): MarkupContent {
+        return this.hoverContent;
+    }
+
+    public hasDocumentation(): boolean {
+        return !!this.documentation || !!this.detail;
+    }
+
     /**
-     * Set the documentation of this object as a `MarkupContent`.
-     * Set the documentation to `null` if empty.
+     * Set the documentation of this object as a `MarkupContent` its detail attribute
+     * as well as its representation when hovering it.
+     * The documentation should always exist and consists at least as the value of detail.
+     *
+     * @param details Some details about this object, as a Markdown string.
      * @param doc The documentation of this object, as a Markdown string.
      *
      * @see MarkupContent
      */
-    public setDocumentation(doc: string): void {
-        if (!doc) {
-            this.documentation = null;
+    public setUserInformations(details: string, doc: string): void {
+        // Set documentation and detail attributes.
+        this.detail = details;
+        this.documentation = !!doc
+            ? {
+                  kind: MarkupKind.Markdown,
+                  value: doc,
+              }
+            : null;
+
+        if (!this.detail && !doc) {
+            return;
+        }
+
+        if (!this.detail) {
+            this.hoverContent = {
```

Comment:
At this point, don’t we have the guarantee that `this.documentation` contains this `{ … }` thingy already? Could we reuse it? Hum not sure, dunno.
All these conditions make me dizzy. It would be great if you could spare a couple of minutes to try to make it simpler or clearer or at least add comments. I’m having a hard time figuring out which symbols can, at each point, be falsey or not, null or not, empty strings or not, etc.